src/test/java/com/es/phoneshop/model/product/ArrayListProductDaoTest.java

```diff
@@ -1,21 +1,78 @@
 package com.es.phoneshop.model.product;
 
+
 import org.junit.Before;
+import org.junit.BeforeClass;
 import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.Spy;
+
+import java.math.BigDecimal;
+import java.util.Arrays;
+import java.util.Currency;
+import java.util.List;
+
 
+import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.when;
 
-public class ArrayListProductDaoTest
-{
-    private ProductDao productDao;
+public class ArrayListProductDaoTest {
+    @Mock
+    private Product product1;
+    @Mock
+    private Product product2;
+
+    @InjectMocks
+    private ArrayListProductDao productDao;
 
     @Before
     public void setup() {
-        productDao = new ArrayListProductDao();
+        MockitoAnnotations.initMocks(this);
```

Comment:
Please make use of @RunWith(MockitoJUnitRunner.class) instead of manual call to initMocks()